kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -69,6 +70,20 @@ TransactionBuilder withTransactionId(
             String applicationId,
             long transactionVersion);
 
+    /**
+     * Set the table properties for the table. When the table already contains the property with
+     * same key, it gets replaced.
+     *
+     * @param engine     {@link Engine} instance to use.
+     * @param properties The table properties to set. These are key-value pairs that can be used to
+     *                   configure the table. And these properties are stored in the table metadata.
+     *
+     * @return updated {@link TransactionBuilder} instance.
+     *
+     * @since 3.2.0
+     */
+    TransactionBuilder withTableProperties(Engine engine, Map<String, String> properties);
+
```

Comment:
We throw the invalid config exceptions as part of the build. Make sure to add them to javadoc as `@throws`. See the existing example `ConcurrentTransactionException`